Updates to draft-ietf-scim-roles-entitlements (#4)
ietf-scim-wg/draft-ietf-scim-roles-entitlements: draft-ietf-scim-roles-entitlements.md

 -
    name: Danny Zollner
    organization: Microsoft

>     email: danny@zollnerd.com 

Reviewer: ```suggestion
    email: danny@zollnerd.com
```

> -

Reviewer: ```suggestion
 -
```

In order to allow for SCIM clients to reduce predictable errors when interacting with SCIM service providers, this document aims to provide a method for SCIM service providers to provide data on what roles and/or entitlements are available, the association between roles and/or entitlements and specific resources so that SCIM clients can consume this data to more efficiently manage resources between directories.

> ### Consuming Roles and Entitlements with SCIM Clients

Reviewer: ```suggestion
## Consuming Roles and Entitlements with SCIM Clients
```

When a SCIM service provider publishes role and entitlement definitions, SCIM clients can consume them efficiently. The process generally follows these steps:

1. Check Provider Support: Check the [ServiceProviderConfig Extension](#serviceproviderconfig-extension) for support for roles and entitlements and the resources associated with roles or entitlements.

> 2. Discover ResourceTypes: Query the /ResourceTypes endpoint to discover which standard and custom role and entitlement [resource types](#resource-type-representation) are supported.

Reviewer: ```suggestion
2. Discover ResourceTypes: Query the `/ResourceTypes` endpoint to discover which standard and custom role and entitlement [resource types](https://datatracker.ietf.org/doc/html/rfc7644#section-4) are supported.
```

                A boolean type that indicates if the SCIM service
                provider supports the "type" sub-attribute for
                the "roles" attribute on the User resource. OPTIONAL.

>            

Reviewer: ```suggestion

```

            types

>                 A multivalue attribute containing list of types supported for "roles" 

Reviewer: ```suggestion
                A multivalue attribute containing list of types supported for "roles"
```

                the "entitlements" attribute on the User resource.
                OPTIONAL.

            types

>                 A multivalue attribute containing list of types supported for "entitlements" 

Reviewer: ```suggestion
                A multivalue attribute containing list of types supported for "entitlements"
```

    type
        A label indicating the role's function.  OPTIONAL

>     

Reviewer: ```suggestion

```

                "description" : "how many users are currently assigned this role, either directly or inherited.",
                "required" : false,
                "caseExact" : false,
                "mutability" : "readOnly",
                "returned" : "default",
                "uniqueness" : "none"
            },
            {
                "name" : "containedBy",
                "type" : "string",
                "multiValued" : true,
                "description" : "A list of "parent" roles that contain a superset of permissions including those granted by this role.",
                "required" : false,
                "caseExact" : false,
                "mutability" : "readOnly",
                "returned" : "default",
                "uniqueness" : "none"
            },
            {
                "name" : "contains",
                "type" : "string",
                "multiValued" : true,
                "description" : "A list of "child" roles that this role grants the rights of.",
                "required" : false,
                "caseExact" : false,
                "mutability" : "readOnly",
                "returned" : "default",

>                 "uniqueness" : "none"   

Reviewer: ```suggestion
                "uniqueness" : "none"
```

                "description" : "how many users are currently assigned this entitlement, either directly or inherited.",
                "required" : false,
                "caseExact" : false,
                "mutability" : "readOnly",
                "returned" : "default",
                "uniqueness" : "none"
            },
            {
                "name" : "containedBy",
                "type" : "string",
                "multiValued" : true,
                "description" : "A list of "parent" entitlement that contain a superset of permissions including those granted by this entitlement.",
                "required" : false,
                "caseExact" : false,
                "mutability" : "readOnly",
                "returned" : "default",
                "uniqueness" : "none"
            },
            {
                "name" : "contains",
                "type" : "string",
                "multiValued" : true,
                "description" : "A list of "child" roles that this entitlement grants the rights of.",
                "required" : false,
                "caseExact" : false,
                "mutability" : "readOnly",
                "returned" : "default",

>                 "uniqueness" : "none"   

Reviewer: ```suggestion
                "uniqueness" : "none"
```
